sonarts-core/src/rules/noIdenticalFunctionsRule.ts

```diff
@@ -89,18 +90,16 @@ export class Rule extends tslint.Rules.AbstractRule {
   }
 }
 
-class Walker extends tslint.RuleWalker {
+class Visitor extends SonarRuleVisitor {
   public functionBlocks: ts.Block[] = [];
 
-  protected visitNode(node: ts.Node): void {
-    if (is(node, ...FUNCTION_LIKE)) {
-      const body = (node as ts.FunctionLikeDeclaration).body;
-      if (is(body, ts.SyntaxKind.Block) && Walker.isBigEnough(body as ts.Block)) {
-        this.functionBlocks.push(body as ts.Block);
-      }
+  public visitFunctionLikeDeclaration(node: ts.FunctionLikeDeclaration) {
+    const { body } = node;
+    if (body && ts.isBlock(body) && Visitor.isBigEnough(body)) {
+      this.functionBlocks.push(body as ts.Block);
```

Comment:
![Code Smell](https://sonarsource.github.io/sonar-github/code_smell.png 'Code Smell') Code Smell: Remove this unnecessary cast. ([typescript:S4325](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=typescript%3AS4325))

[See it in SonarQube](https://next.sonarqube.com/sonarqube/project/issues?id=sonarts&branch=sonar-issue&issues=AWCd632l_Q-0dz4JAbfL&open=AWCd632l_Q-0dz4JAbfL)

[ISSUE_KEY]: AWCd632l_Q-0dz4JAbfL